CONTRIBUTING.md

```diff
@@ -61,7 +61,20 @@ When you are contributing to changelog, please follow these suggestions:
 
 ### Testing
 
-Tests are stored in [tests](/tests) directory.
+Tests are stored in [tests](/tests) directory:
+
+- `tests/unit`
+    - testing small units/parts of the code
+    - strictly offline
+- `tests/integration`
+    - testing bigger parts of codebase (integration between multiple units, packit python API)
+    - use [requre](https://github.com/packit-service/requre)
+      for remote communication => offline in the CI
+    - prefer [requre](https://github.com/packit-service/requre) instead of mocking
+- `tests/functional`
+    - testing packit as a CLI
+    - be careful what you run -- no requre, no mocking
+
```

Comment:
As @jscotka suggested, we can split (non)requre tests:

Reply:
a)
- `tests/integration-requre` - requre, avoid mocking
- `tests/integration` - without requre, flexmock

Reply:
b) current state:
- `tests/testsuite_basic`
  - `unit`
  - `integration`
  - `functional`
- `tests/testsuite_recording`
  - `unit`
  - `integration`

Reply:
After today's discussion with Honza, I will go with the first one and make the unit tests strictly offline.

Reply:
I could not make it in that way, current structure:

- `tests/`
  - `unit`
  - `integration`
  - `functional`
- `tests_recording`